### PR TITLE
cleanup unused options & support per-level compression

### PR DESCRIPTION
**Pull request by bobotu**

### CODE REVIEW

File: table/table_test.go [hunk 1]
```diff
@@ -45,11 +45,11 @@ var defaultBuilderOpt = options.TableBuilderOptions{
 		BitsPerKeyHint: 40,
 		RealSuffixLen:  10,
 	},
-	Compression:         options.ZSTD,
+	CompressionPerLevel: []options.CompressionType{options.ZSTD},
 	BlockSize:           4 * 1024,
 	HashUtilRatio:       0.75,
 	WriteBufferSize:     1024 * 1024,
-	MaxLevels:           7,
+	MaxLevels:           1,
```
Comment (coocood) [inline table/table_test.go]: Why change this?

Comment (coocood) [reply table/table_test.go]: The table size is small, we can still write multiple level tables.

Comment (bobotu) [reply table/table_test.go]: Tests in `table` package only test on one tables, and has no concept about level.

Comment (bobotu) [reply table/table_test.go]: Tests in table package only test on one table, and has no concept about level.

**Comment by coocood** [comment]
LGTM

**Pull request merged**